packages/headless/src/controllers/triggers/headless-query-trigger.test.ts

```diff
@@ -0,0 +1,86 @@
+import {QueryTrigger, buildQueryTrigger} from './headless-query-trigger';
+import {
+  buildMockSearchAppEngine,
+  MockSearchEngine,
+} from '../../test/mock-engine';
+import {triggers, query} from '../../app/reducers';
+import {updateQuery} from '../../features/query/query-actions';
+import {executeSearch} from '../../features/search/search-actions';
+
+describe('QueryTrigger', () => {
+  let engine: MockSearchEngine;
+  let queryTrigger: QueryTrigger;
+
+  function initQueryTrigger() {
+    queryTrigger = buildQueryTrigger(engine);
+  }
+
+  function getUpdateQueryAction() {
+    return engine.actions.find((a) => a.type === updateQuery.type);
+  }
+
+  function registeredListeners() {
+    return (engine.subscribe as jest.Mock).mock.calls.map((args) => args[0]);
+  }
+
+  beforeEach(() => {
+    engine = buildMockSearchAppEngine();
+    initQueryTrigger();
+  });
+
+  it('initializes', () => {
+    expect(queryTrigger).toBeTruthy();
+  });
+
+  it('it adds the correct reducers to the engine', () => {
+    expect(engine.addReducers).toHaveBeenCalledWith({
+      triggers,
+      query,
+    });
+  });
+
+  it('exposes a #subscribe method', () => {
+    expect(queryTrigger.subscribe).toBeTruthy();
+  });
+
+  it('when the #engine.state.triggers.query is not updated, it does not dispatch #updateQuery, #executeSearch, and #logQueryTrigger', () => {
```

Comment:
It's usually better to leverage a describe + beforeEach block & split each of these expectation in their own "it" block.

This way when one fails, it's easily identifiable.

Applies for other tests